xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/tests/unit/utilities/XWikiIcon.spec.js

```diff
@@ -0,0 +1,105 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+import XWikiIcon from "../../../utilities/XWikiIcon";
+import {mount} from '@vue/test-utils'
+import _ from "lodash";
+import $ from 'jquery';
+import Vue from "vue";
+
+global.$ = global.jQuery = $;
+
+function initWrapper(params = {}) {
+  return mount(XWikiIcon, _.merge({
+    propsData: {
+      iconDescriptor: {}
+    }
+  }, params));
+}
+
+describe('XWikiIcon.vue', () => {
+  it('Displays an icon using a font', () => {
+    const wrapper = initWrapper({
+      propsData: {
+        additionalClasses: ["newClass"],
+        iconDescriptor: {
+          cssClass: "fa fa-plus"
+        }
+      }
+    });
+
+    expect(wrapper.html()).toBe('<span class="fa fa-plus newClass"></span>');
```

Comment:
I'm not sure it is a really good practice to evaluate the whole html of the component, this seems very fragile to me.
I would rather focus the tests on what's being actually tested, in that case that the [class](https://vue-test-utils.vuejs.org/api/wrapper/#classes) attribute:
```js
expect(wrapper.classes()).toContain('fa')
expect(wrapper.classes()).toContain('fa-plus')
expect(wrapper.classes()).toContain('newClass')
```
So that if one day we decide to change the tag name, or add an attribute to the element, we won't have to update the tests.

Reply:
I don't think we have a rule for that. When the html of the component is small, I find this kind of assertions easier to read and more exhaustive.
What you propose does not assert:
- the kind of produced tag
- the absence of any other attribute

Of course whenever the html is getting larger, this does not work, or only by first selecting sub elements, for instance `expect(wrapper.find('.someclass).html()).toBe('...')`

Reply:
> What you propose does not assert:
>
> - the kind of produced tag
> - the absence of any other attribute

Yes, this is exactly my point!

The test is testing: "Displays an icon using a font", so what it should assert is whether the icon is displayed on the component. It should not care about the other component attributes or the component tag, it is not part of its concern.

If the element tag is really important though, you can create another test, that is going to assert that specifically:
```js
  it('Has correct tag', () => {
    const wrapper = initWrapper({ /* ... */ });

    expect(wrapper.name()).toBe('span')
  });
```

But actually the [`name` method is deprecated](https://vue-test-utils.vuejs.org/api/wrapper/#name) because this is considered a [bad practice](https://vue-test-utils.vuejs.org/upgrading-to-v1/#name) to go this far into implementation details.

Reply:
> What you propose does not assert:
>
> - the kind of produced tag
> - the absence of any other attribute

Yes, this is exactly my point!

The test is testing: "Displays an icon using a font", so what it should assert is whether the icon is displayed on the component. It should not care about the other component attributes or the component tag, it is not part of its concern.

If the element tag is really important though, you can create another test, that is going to assert that specifically:
```js
  it('Has correct tag', () => {
    const wrapper = initWrapper({ /* ... */ });

    expect(wrapper.name()).toBe('span')
  });
```

But actually the [`name` method is deprecated](https://vue-test-utils.vuejs.org/api/wrapper/#name) because this is considered a [bad practice](https://vue-test-utils.vuejs.org/upgrading-to-v1/#name) to go this far into implementation details.